src/components/NewContactPage.jsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+function NewContactPage({ contacts, setContacts }) {
+  const [newContact, setNewContact] = useState({
+    firstName: "",
+    lastName: "",
+    street: "",
+    city: "",
+  });
+
+  const navigate = useNavigate();
+
+  const handleFormUpdate = (event) => {
+    //Load all data from this input
+    const { id, value } = event.target;
+    //Check what text has been updated and update the state accordingly
+    if (id === "firstName") setNewContact({ ...newContact, firstName: value });
+    else if (id === "lastName")
+      setNewContact({ ...newContact, lastName: value });
+    else if (id === "city") setNewContact({ ...newContact, city: value });
+    else if (id === "street") setNewContact({ ...newContact, street: value });
+  };
+  const postRequestOptions = {
+    method: "POST",
+    headers: {
+      "Content-Type": "application/json",
+    },
+    body: JSON.stringify(newContact),
```

Comment:
There's a potential issue with the way you're using newContact in your postRequestOptions. You define postRequestOptions outside of handleSubmit, which means it captures the initial state of newContact at the time of component initialization, not when the form is submitted...